Headers/Core/Engine/Engine.h

```diff
@@ -23,7 +24,7 @@ class Engine {
     void update( double delta );
     void render();
     
-    void consumeEvents( std::vector<Event*> * );
+    void consumeEvents( std::vector<std::shared_ptr<Event>> * );
```

Comment:
How is responsible for deleting the events each frame? If we call clear while other shared pointers still point to these objects they may never be release and we may be creating a memory leak.